### ISSUE TITLE
Misunderstanding of the validation table

### ISSUE THREAD

**Issue by 1fanliu** [robin-chan/meta-ood#3]
Dear Robin,

I've run your model based on the provided checkpoint, but the results are far below the one in the paper. However, it matches the SegmentMeIfYouCan's results. For example, it shows 17.4 FPR in SegmentMeAnomaly, while _**it only has 77.42 FPR in Fishyscapes lost and found.**_

The issue is when people compare the results between different validation sets, they only utilise 1 checkpoint instead of handpicking the best one. As in real practice, it makes none-sense to utilise different sensors to capture the views among different scenes.

What you did is very funny and unacceptable. 

Could you please release the original checkpoint that was used in your meta-ood paper? 

Regards,
Fan

**Comment by robin-chan** [comment]
Dear Fan,

I am sorry to hear that my code has caused you troubles. It is for sure not my intention to provide any wrong numbers.

Regarding your issue, I used the same checkpoint for all experiments across different datasets. In case the provided link is faulty, here is a line to another copy (for the DeeplabV3+ model): 

- https://uni-wuppertal.sciebo.de/s/kCgnr0LQuTbrArA/download

The latter checkpoint has been evaluated on multiple datasets. In the following, I want to report the results for entropy maximization and explain how these numbers are produced:

1. **LostAndFound Test**: AP 76.45%, FPR95 9.58%, computed using this repo [evaluation.py](https://github.com/robin-chan/meta-ood/blob/master/evaluation.py)
2. **LostAndFound Static Validation Data**: AP 81.42%, FPR95 5.02%, computed using this repo [evaluation.py](https://github.com/robin-chan/meta-ood/blob/master/evaluation.py). The 30 validation images with labels can be downloaded here https://fishyscapes.com/dataset (needs to be preprocessed though) 
3. **Fishyscapes Static**: AP 86.55%, FPR95 8.55%, submitted to Fishyscapes Benchmark. See also   https://fishyscapes.com/results under _anonymous submission (ruby)_
4. **Fishyscapes Lost & Found**:  AP 29.96%, FPR95 35.14%, submitted to Fishyscapes Benchmark. See also   https://fishyscapes.com/results under _anonymous submission (ruby)_
5. **SegmentMeIfYouCan LostAndFound NoKnown**: AP 77.90%, FPR95 9.70%, can be reproduced with the SegmentMeIfYouCan benchmark suite. See also https://github.com/SegmentMeIfYouCan/road-anomaly-benchmark/blob/master/some_methods_inference_public.py
6. **SegmentMeIfYouCan RoadAnomaly21**: AP 85.47%, FPR95 15%, submitted to SegmentMeIfYouCan benchmark

I hope that this makes the results more transparant.

Best,
Robin

**Comment by 1fanliu** [comment]
The checkpoint you newly provided still cannot match the result in your meta-ood paper.  For example, your checkpoint results are **0.12960443 (fpr)** and **0.09576 (fpr)** in fishyscape (static) and lost & found. They are both higher than the reported number in the related table, which are 0.05 and 0.09, respectively.

**Comment by robin-chan** [comment]
Dear Fan,

please note that in the paper the results are reported as decimals, ie $0.0958 = 9.58$% for LostAndFound Test. (This is what I did in the early stages of writing papers, sorry for the confusion)
Besides, the labels for the Fishyscapes static dataset are private and the results can only be obtained by submitting to the Fishyscapes Benchmark. As written in the previous comment, the evaluation yields FPR95 of $8.55$% $= 0.0855 \approx 0.09$, which I also report in the paper, see table 1. 
The FPR of 0.05 (=5%) you are referring to corresponds to the Fishyscapes Static Validation data. This result of course depends on how the images are extracted from the Fishyscapes Tensorflow implementation. If you want to reproduce these numbers, I can send you the data I used for the experiments.

Best,
Robin

![image](https://user-images.githubusercontent.com/46135971/180395805-5f8d56d3-d5ef-4ba2-b1ec-bd4ba34bd9ce.png)

**Comment by 1fanliu** [comment]
Dear Robin,

Thanks for all your declaration. Now I figure out everything, and I apologize for the strong words before. Thanks for your time and your explanation.

For the fishycape (static), I grab from here (https://github.com/giandbt/synboost), and the different extraction causes the worse FPR. (If it is random, why do people compare it?) 

I've organized the table of the validation results below, and it will be easier for the people who are willing to follow your work.
| metrics  | fishy (static) | fishy (l&f) | smiuc (anomaly) | smiuc (obstacle) | l&f |
   |:--------: |:-----:   |:-----:   |:-----:   |:-----: |:-----:|
| roc | 0.9714000657372416     | 0.9279047566902856   | 0.9589541873990705   | 0.9991700848756975   | 0.9808166044268044
| fpr | 0.12960443386222215 | 0.37221749787404335  | 0.1743236753641748   | 0.004091882265842287     | 0.0957650095544906
| ap  | 0.7248069822617257     | 0.40800855463031144  | 0.8012561465099882   | 0.9414367836483518   | 0.7644881492358063


Best regards,
Fan
